dask/dataframe/io.py

```diff
@@ -441,8 +441,12 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
 
     dask_kwargs = dask_kwargs or {}
 
-    DataFrame._get(merge(df.dask, dsk), (name, df.npartitions - 1),
-                   get=get, **dask_kwargs)
+    dsk = merge(df.dask, dsk)
+    keys = (name, df.npartitions - 1)
+    if compute:
+        DataFrame._get(dsk, keys, get=get, **dask_kwargs)
+    else:
+        return dsk, keys
```

Comment:
Thoughts on returning a dask.Delayed value instead of an explicit dask graph?

I see that this is what is already done in `to_castra`.  It's good to be consistent (as this is) so we should probably either change both or neither.